### PR TITLE
[lldb] Add lldbutil.install_to_target() helper

### PR DESCRIPTION
**Pull request by slydiman**
It can be used in tests #91918, #91931 and such.

### CODE REVIEW

File: lldb/packages/Python/lldbsuite/test/lldbutil.py [hunk 1]
```diff
@@ -1654,6 +1654,26 @@ def find_library_callable(test):
     )
 
 
+def target_install(test, filename=None, path=None):
+    test.assertTrue(filename or path, "filename or path must be specified.")
+    if filename is None:
+        filename = os.path.basename(path)
+    if path is None:
+        path = test.getBuildArtifact(filename)
```
Comment (labath) [inline lldb/packages/Python/lldbsuite/test/lldbutil.py:1662]: This seems a bit... magical. Can we just have the caller specify a full path to the thing it wants to install? `self.getBuildArtifact` is not that much extra typing.

File: lldb/packages/Python/lldbsuite/test/lldbutil.py [hunk 2]
```diff
@@ -1654,6 +1654,26 @@ def find_library_callable(test):
     )
 
 
+def target_install(test, filename=None, path=None):
```
Comment (labath) [inline lldb/packages/Python/lldbsuite/test/lldbutil.py:1657]: This is optional, but I think something like `install_to_target` would be a better match for the naming style of the surrounding functions.

File: lldb/packages/Python/lldbsuite/test/lldbutil.py (hunk 1 above)
Comment (slydiman) [reply lldb/packages/Python/lldbsuite/test/lldbutil.py:1662]: I expected the following usage:

1: Use filename
```
target_path = lldbutil.install_to_target(self, "a.out")
```
2: Use path
```
local_path = self.getBuildArtifact("some_file")
# write to local_path here 
...
# avoid typing "some_file" constant twice, just extract the filename from the local_path
target_path = lldbutil.install_to_target(self, path=local_path) 
```
3: Use both
```
# "temp" is the filename on the remote target
target_path = lldbutil.install_to_target(self, "temp", local_temp_path_outside_build_dir)
```

File: lldb/packages/Python/lldbsuite/test/lldbutil.py (hunk 2 above)
Comment (slydiman) [reply lldb/packages/Python/lldbsuite/test/lldbutil.py]: Updated. Thanks.

File: lldb/packages/Python/lldbsuite/test/lldbutil.py (hunk 1 above)
Comment (slydiman) [reply lldb/packages/Python/lldbsuite/test/lldbutil.py]: Ok. I kept only the full path parameter. Updated. Thanks.

**Review by labath — APPROVED** [review]

**Pull request merged**